docs/docs/api/apiReference.md

```diff
@@ -0,0 +1,303 @@
+
+Parameter used to configure animation. Can be set up for the entire app or selectively for a particular notification. <br/>
+Below is the global configuration:
+
+```tsx
+import { createNotifications, RotateInRotateOut } from 'react-native-notification'
+
+const { NotificationsProvider } = createNotifications({
+    animationConfig: RotateInRotateOut
+})
+```
+
+And can be overwrite in local scope, when calling notification:
+
+```tsx
+import { SlideInLeftSlideOutRight } from 'react-native-notification'
+
+[...]
+
+notify('success', {
+  params: {
+    title: 'Success',
+    description: 'This is where the toast text goes',
+  },
+  config: {
+    animationConfig: SlideInLeftSlideOutRight,
+  },
+})
+```
+We've prepared for you a couple of ready-to-use animations (just import them from the library):
+
+- `RotateZIn`
+- `ZoomInDownZoomOutUp`
+- `ZoomInDownZoomOutDown`
+- `RotateInRotateOut`
+- `SlideInLeftSlideOutRight`
+
+***
+### `createNotifications()`
+
+API is used to initialize the library in the project.
+
+```tsx
+import { createNotifications } from 'react-native-notification'
+
+const { NotificationsProvider } = createNotifications()
+```
+The `createNotifications()` invoked without any configuration, as in the above example, is fairly enough to use the library on its basic level.<br/>
+Here is an example of all available parameters, that can be adjusted in `createNotifications()`:
+
+```tsx
+const { NotificationsProvider } = createNotifications({
+  duration: number // -> default 3000
+  notificationPosition: NotificationPosition
+  animationConfig: CustomAnimationConfig
+  gestureConfig: GestureConfig
+  isNotch?: boolean // -> by default uses react-native-device-info
+  defaultStylesSettings: {
+    darkMode: false,
+    globalConfig: StyleProps
+    successConfig: StyleProps
+    errorConfig: StyleProps
+    warningConfig: StyleProps
+    infoConfig: StyleProps
+  },
+  variants: CustomVariants
+})
+```
+Check on [`StyleProps`](http://localhost:3000/react-native-notificated/docs/api/types#styleprops)
+
+The `defaultStylesSettings` provides styles for all default notifications. `globalConfig` is on the top of the hierarchy, and distributes styles for all types of build-in notifications: success, error, warning, and info. For example `successConfig` (if provided) will overwrite `globalConfig` styles only for notifications of type 'success'. `successConfig` styles can be overwritten on the lowest level when calling notification 'success' using a `notify()`. Check on [`Order of settings overwriting`](https://thewidlarzgroup.github.io/react-native-notificated/docs/comprehensive-configuration/order-of-settings-overwriting) for more details.
+
+***
+### `generateAnimationConfig()`
+
+This method allows to set up custom animations. <br/>
+
+```tsx
+generateAnimationConfig(config: CustomAnimationConfig)
+```
+Check on [`CustomAnimationConfig`](http://localhost:3000/react-native-notificated/docs/api/types#customanimationconfig)
```

Comment:
```suggestion
Check on [`CustomAnimationConfig`](XXX/docs/api/types#customanimationconfig)
```
we should put a proper url for domain where the docs are hosted (thewidlarzgroup.github.io/.....)

Reply:
it would need to be updated for all the links added in these new sections

Reply:
\+ in this particular place, I would point the developer also to this part of the documentation: `animations/custom-transitions`

Reply:
url updated and added additional reference to docs